packages/api/src/task/scheduler.ts

```diff
@@ -264,6 +277,36 @@ export class TaskScheduler {
     await this.updateTask(task, { status });
   }
 
+  async retryTask(task: WithID<Task>, event?: messages.TaskResult) {
+    let attempts = task.status.attempts + 1 ?? 1;
+    if (attempts <= MAX_RETRY_ATTEMPTS) {
+      const status: Task["status"] = {
+        ...task.status,
+        attempts: attempts,
+      };
+
+      await this.updateTask(task, { status });
+      task.status = status;
+
+      if (attempts > 1) {
+        // No timeout at first retry, 15 seconds * attempt after that
+        let retryDelay = attempts * TASK_RETRY_DELAY_COEFFICIENT;
+        await new Promise((resolve) => setTimeout(resolve, retryDelay));
+      }
+
+      await this.enqueueTask(task);
+      return true;
```

Comment:
I don't think these return values are used. If really not, should remove (`return false` below as well). For a moment I was confused thinking this was the "ack or nack the event" boolean value, but it is being completely ignored.

So the other rule of thumb here would be to not have "extraneous" inputs nor outputs to the code. It makes things look like they do other things. If we ever do need to know on the call-site if the task was really retried or not, we can add these return values back.